cmd/kubeadm/app/phases/uploadconfig/uploadconfig_test.go

```diff
@@ -51,6 +53,11 @@ func TestUploadConfiguration(t *testing.T) {
 			} else if masterCfg.Data[kubeadmconstants.MasterConfigurationConfigMapKey] == "" {
 				t.Errorf("Fail to find ConfigMap key = %v", err)
 			}
+
+			configStr := masterCfg.Data[kubeadmconstants.MasterConfigurationConfigMapKey]
```

Comment:
I rewrote this test so I wouldn't have to also cherry pick https://github.com/kubernetes/kubernetes/pull/52109 (which is pretty intertwined with the actual new test in #53559).

Is that the right approach? Should I just redo this and cherry pick  https://github.com/kubernetes/kubernetes/pull/52109 as well?

Reply:
IMO the risk of going ahead without the unit test is acceptable in this specific case, but being the PR writer it is better to have a second opinion if possible...

Reply:
I think I covered this new check reasonably with the test I added. I just want to make sure I'm not making future cherry picks even harder by diverging more than necessary.